lib/Conversion/TorchOnnxToTorch/ComMicrosoftDomain.cpp

```diff
@@ -63,4 +63,393 @@ void mlir::torch::onnx_c::populateComMicrosoftDomain(
+            rewriter.create<Torch::PrimListConstructOp>(
+                binder.getLoc(),
+                Torch::ListType::get(Torch::IntType::get(query.getContext())),
+                llvm::SmallVector<Value>{cstBatchSize, cstNumHeads,
+                                         cstSequenceLength, cstHeadSize});
+        Value qInput = rewriter.create<Torch::AtenReshapeOp>(
+            loc,
+            queryType.getWithSizesAndDtype(queryReshapeSizesInt,
+                                           queryType.getOptionalDtype()),
+            query, queryReshapeSizesList);
+
+        // Reshaping key.
+        SmallVector<int64_t, 6> kvReshapeSizesInt{batchSize, kvNumHeads,
+                                                  sequenceLength, headSize};
+        Value kvReshapeSizesList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(query.getContext())),
+            llvm::SmallVector<Value>{cstBatchSize, cstKVNumHeads,
+                                     cstSequenceLength, cstHeadSize});
+        Torch::ValueTensorType keyType =
+            cast<Torch::ValueTensorType>(key.getType());
+        Value kInput = rewriter.create<Torch::AtenReshapeOp>(
+            loc,
+            keyType.getWithSizesAndDtype(kvReshapeSizesInt,
+                                         keyType.getOptionalDtype()),
+            key, kvReshapeSizesList);
+
+        // Reshaping value.
+        Torch::ValueTensorType valueType =
+            cast<Torch::ValueTensorType>(value.getType());
+        Value vInput = rewriter.create<Torch::AtenReshapeOp>(
+            loc,
+            valueType.getWithSizesAndDtype(kvReshapeSizesInt,
+                                           valueType.getOptionalDtype()),
+            value, kvReshapeSizesList);
+
+        Value cstNone = rewriter.create<Torch::ConstantNoneOp>(loc);
+        Value cstFalse = rewriter.create<Torch::ConstantBoolOp>(loc, false);
+
+        Value qRotary = qInput, kRotary = kInput;
+        if (doRotary) {
+          // `totalSequenceLength` is a scalar tensor.
+          Value scalarTotalSeqLens = rewriter.create<Torch::AtenItemOp>(
+              loc, rewriter.getType<Torch::IntType>(), totalSequenceLength);
+          Value cstIntOne = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(1));
+          Type boolTy = rewriter.getType<Torch::BoolType>();
+          Value condA = rewriter.create<Torch::AtenGtIntOp>(
+              loc, boolTy, cstSequenceLength, cstIntOne);
+          Value condB = rewriter.create<Torch::AtenNeIntOp>(
+              loc, boolTy, cstSequenceLength, scalarTotalSeqLens);
+          //   if (sequence_length > 1 && sequence_length !=
+          //   total_sequence_length)
+          //         is_subsequent_prompt = false;  // Subsequent prompt
+          Value isSubsequentPrompt = rewriter.create<Torch::Aten__And__BoolOp>(
```

Comment:
why flip [is_first_prompt ](https://github.com/microsoft/onnxruntime/blob/65008cbb7393b121400a40dd8af4cc93d506918f/onnxruntime/contrib_ops/cpu/bert/group_query_attention.cc#L133C20-L133C35)to `isSubsequentPrompt ` ?